### DISCUSSION TITLE
Slack channel for community

### DISCUSSION THREAD

**Discussion by shahzadmajeed** [pointfreeco/swift-composable-architecture#1118] [General]
Hi, I am wondering if you have a slack channel for TCA community to engage and discuss problems, solutions and ideas? If not, could we create one?

**Comment by tgrapperon** [comment]
Hello @shahzadmajeed!
I'm not aware of such channel, but I'm wondering if there is something in Slack that GitHub discussions can't provide. Do you see something missing?

**Comment by shahzadmajeed** [comment]
IMO, collaborative discussions are easier, there is more visibility, quick
responses are expected. Maybe it’s just me 🤷‍♂️

On Wed, Jun 1, 2022 at 4:08 PM Thomas Grapperon ***@***.***>
wrote:

> Hello @shahzadmajeed <https://github.com/shahzadmajeed>!
> I'm not aware of such channel, but I'm wondering if there is something in
> Slack that GitHub discussions can't provide. Do you see something missing?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pointfreeco/swift-composable-architecture/discussions/1118#discussioncomment-2866738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJHI427KQA2BI4VEXE7V43VM67K5ANCNFSM5XSILVXQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <pointfreeco/swift-composable-architecture/repo-discussions/1118/comments/2866738
> @github.com>
>
-- 
Sent from my iPhone.

**Comment by mbrandonw** [comment]
We're definitely open to it, and in fact there is a private slack called "iOS folks" that has a pretty active TCA channel, but sadly the slack is invite-only and so a pain to add people.

We'd be down to start a new one to make it easier, but the costs of running a Slack are quite prohibitive (~$7/user/month?!), and Stephen and I personally don't use discord so rather not manage that.

**Reply by tgrapperon** [reply]
The free version of Slack limits the search to the latest 10000 messages. There are other limitations, like access to some administrative tools or the number of linked applications, but I don't think this would be problematic for this kind of semi-public/open setup.

**Reply by stephencelis** [reply]
It'd be really sad to lose discussion about various topics over time, though, especially given the learning nature of Point-Free. 😞

**Reply by tgrapperon** [reply]
I agree, I was just mentioning the existence of the free plan that many channels are using out there and which impedes the overall experience.

Maybe you found your next huge project after isowords!

**Reply by shahzadmajeed** [reply]
Thank you for your replies. 

I definitely forgot the unpaid version limitations and don't like it either. It causes lot of repeating questions when people can't find relevant content. iOS Developers group is an example of that.

My main concern is that TCA adoption involves some learning curve which could be reduced a bit. Here are other ways of doing that:

- Improve documentation
- Add more fixture projects covering more common cases (I can provide you with more use cases if you want. Even help with creating few projects but I would need your help)
- Simple UIKit examples (I see those missing currently. I know isowords have some UIKit examples)
- Simple higher order functions examples

I'm suggesting all this because I see TCA as a great alternative for current modern architectures (most of them are missing state management) but we need to reduce the friction in adopting TCA first.

**Reply by shahzadmajeed** [reply]
And I also want to thank you for open sourcing TCA. That is a great help to iOS community (I know people from Android trying to adopt these concepts as there is a TCA implementation for Android as well)
